packages/web-runtime/src/helpers/language.ts

```diff
@@ -16,3 +18,29 @@ export const setCurrentLanguage = ({
     document.documentElement.lang = currentLanguage
   }
 }
+
+/**
+ * Loads all app translations for one given language.
+ * This should be called each time the language is being changed.
+ */
+export const loadAppTranslations = ({
+  apps,
+  gettext,
+  lang
+}: {
+  apps: Record<string, ApplicationInformation>
+  gettext: Language
+  lang: string
+}) => {
+  const appTranslations: Translations = {}
+  Object.values(apps).forEach((app) => {
+    const { translations } = app
+    if (gettext.translations[lang] && translations[lang]) {
```

Comment:
What is the `gettext.translations[lang]` check for here? 🙈

Reply:
As discussed: It checks if we support the requested language and that there are translations available for it. This is basically determined by the core translations.